gems/pending/OpenStackExtract/MiqOpenStackVm/MiqOpenStackInstance.rb

```diff
@@ -30,7 +30,7 @@ def instance
   end
 
   def snapshot_metadata
-    @snapshot_metadata ||= instance.metadata.get(:miq_snapshot)
+    @snapshot_metadata ||= instance.metadata.length > 0 && instance.metadata.get(:miq_snapshot)
```

Comment:
The Excon::Errors::NotFound is indirectly derived from Error, instead of StandardError. which can be rescued by default. So the rescue way wouldn't prevent these error messages from our log files.

Will keep the original solution: check length > 0 before querying it.